controllers/medusa/medusabackupschedule_controller.go

```diff
@@ -0,0 +1,141 @@
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package medusa
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+
+	medusav1alpha1 "github.com/k8ssandra/k8ssandra-operator/apis/medusa/v1alpha1"
+	cron "github.com/robfig/cron/v3"
+)
+
+type Clock interface {
+	Now() time.Time
+}
+
+// MedusaBackupScheduleReconciler reconciles a MedusaBackupSchedule object
+type MedusaBackupScheduleReconciler struct {
+	client.Client
+	Scheme *runtime.Scheme
+	Clock  Clock
+}
+
+type RealClock struct{}
+
+func (r *RealClock) Now() time.Time {
+	return time.Now()
+}
+
+//+kubebuilder:rbac:groups=medusa.k8ssandra.io,namespace="k8ssandra",resources=medusabackupschedules,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups=medusa.k8ssandra.io,namespace="k8ssandra",resources=medusabackupschedules/status,verbs=get;update;patch
+//+kubebuilder:rbac:groups=medusa.k8ssandra.io,namespace="k8ssandra",resources=medusabackupschedules/finalizers,verbs=update
+// +kubebuilder:rbac:groups=medusa.k8ssandra.io,namespace="k8ssandra",resources=cassandrabackups,verbs=get;list;watch;create
+
+func (r *MedusaBackupScheduleReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	logger := log.FromContext(ctx).WithValues("medusabackupscheduler", req.NamespacedName)
+
+	backupSchedule := &medusav1alpha1.MedusaBackupSchedule{}
+	err := r.Get(ctx, req.NamespacedName, backupSchedule)
+	if err != nil {
+		return ctrl.Result{}, client.IgnoreNotFound(err)
+	}
+
+	sched, err := cron.ParseStandard(backupSchedule.Spec.CronSchedule)
+	if err != nil {
+		// The schedule is in incorrect format
+		return ctrl.Result{}, err
+	}
+
+	previousExecution, err := getPreviousExecutionTime(ctx, backupSchedule)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	now := r.Clock.Now().UTC()
+
+	// Calculate the next execution time
+	nextExecution := sched.Next(previousExecution).UTC()
+
+	createBackup := false
+
+	if nextExecution.Before(now) {
+		nextExecution = sched.Next(now)
+		previousExecution = now
+		createBackup = true && !backupSchedule.Spec.Disabled
+	}
+
+	// Update the status if there are modifications
+	if backupSchedule.Status.LastExecution.Time.Before(previousExecution) ||
+		backupSchedule.Status.NextSchedule.Time.Before(nextExecution) {
+		schedulePatch := client.MergeFrom(backupSchedule.DeepCopy())
+		backupSchedule.Status.NextSchedule = metav1.NewTime(nextExecution)
+		backupSchedule.Status.LastExecution = metav1.NewTime(previousExecution)
+
+		if err := r.Client.Patch(ctx, backupSchedule, schedulePatch); err != nil {
+			return ctrl.Result{}, err
+		}
+	}
+
+	if createBackup {
+		// TODO Verify here no previous jobs is still executing?
```

Comment:
We may need to add a check right now because I've tested your branch and it tends to create two backups almost at the same times very often:
<img width="1010" alt="Capture d’écran 2022-06-07 à 12 20 47" src="https://user-images.githubusercontent.com/5096002/172360652-3d986a62-1098-4b49-8e4f-778c9ca19440.png">

Do you see how we could protect ourselves from those duplicates?

Reply:
What cron schedule did you use?

Reply:
But I have an idea why that happens..

Reply:
The latest commit has status updates filtered out, which I think caused the duplicate behavior. Although perhaps we could still get unnecessary invocations and read from a stale cache..